charts/newrelic-k8s-metrics-adapter/templates/apiservice/job-patch/job-createSecret.yaml

```diff
@@ -3,48 +3,60 @@ apiVersion: batch/v1
 kind: Job
 metadata:
   namespace: {{ .Release.Namespace }}
-  name: {{ template "newrelic-k8s-metrics-adapter.fullname" . }}-apiservice-create
+  name: {{ template "common.naming.fullname" . }}-apiservice-create
   annotations:
     "helm.sh/hook": pre-install,pre-upgrade
     "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
   labels:
-    app: {{ template "newrelic-k8s-metrics-adapter.name" $ }}
-{{ include "newrelic-k8s-metrics-adapter.labels" $ | indent 4 }}
+    app: {{ template "common.naming.name" . }}
+    {{- include "common.labels" . | nindent 4 }}
 spec:
   template:
     metadata:
-      name: {{ template "newrelic-k8s-metrics-adapter.fullname" . }}-apiservice-create
+      name: {{ template "common.naming.fullname" . }}-apiservice-create
       labels:
-        app: {{ template "newrelic-k8s-metrics-adapter.name" $ }}
-{{ include "newrelic-k8s-metrics-adapter.labels" $ | indent 8 }}
+        app: {{ template "common.naming.name" . }}
+        {{- include "common.labels" . | nindent 8 }}
     spec:
+      {{- with include "common.images.renderPullSecrets" ( dict "pullSecrets" (list .Values.image.pullSecrets) "context" .) }}
+      imagePullSecrets:
+        {{- . | nindent 8 }}
+      {{- end }}
       containers:
         - name: create
-          image: {{ .Values.apiServicePatchJob.image.repository }}:{{ .Values.apiServicePatchJob.image.tag }}
+          image: {{ include "common.images.image" ( dict "imageRoot" .Values.apiServicePatchJob.image "context" .) }}
           imagePullPolicy: {{ .Values.apiServicePatchJob.image.pullPolicy }}
           args:
             - create
-            - --host={{ template "newrelic-k8s-metrics-adapter.fullname" . }},{{ template "newrelic-k8s-metrics-adapter.fullname" . }}.{{ .Release.Namespace }}.svc
+            - --host={{ template "common.naming.fullname" . }},{{ template "common.naming.fullname" . }}.{{ .Release.Namespace }}.svc
             - --namespace={{ .Release.Namespace }}
-            - --secret-name={{ template "newrelic-k8s-metrics-adapter.fullname" . }}-apiservice
+            - --secret-name={{ template "common.naming.fullname" . }}-apiservice
             - --cert-name=tls.crt
             - --key-name=tls.key
-          {{- if .Values.apiServicePatchJob.image.volumeMounts }}
+          {{- with .Values.apiServicePatchJob.extraEnv }}
+          env:
+          {{- toYaml . | nindent 10 }}
+          {{- end }}
+          {{- with .Values.apiServicePatchJob.extraEnvFrom }}
+          envFrom:
+          {{- toYaml . | nindent 10 }}
+          {{- end }}
+          {{- with .Values.apiServicePatchJob.extraVolumeMounts }}
```

Comment:
Do we need this snippet? As this is going to be a short-lived job I'm mildly reluctant to add features to it.

Reply:
I have removed the env and ExtraEnv and left only the Volumes and VolumeMounts since they were already present in the previous chart.